src/Compilers/CSharp/Portable/Symbols/Synthesized/GeneratedLabelSymbol.cs

```diff
@@ -9,9 +9,19 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
     internal sealed class GeneratedLabelSymbol : LabelSymbol
     {
+        private readonly string _name;
+
         public GeneratedLabelSymbol(string name)
-            : base(LabelName(name))
```

Comment:
I think generated label has name only to observe it in debugger, where it could be helpful. 
I do not think these names actually get emitted in any form.

Source label names are observable (via semantic model, I think), but creating the names eagerly was causing a good amount of allocations in batch compile scenario where names are unused. 
I made LabelSymbol to not require name eagerly and this is a sideeffect of that